kafka-avro-adapter/src/cs_producer.cpp

```diff
@@ -18,140 +18,177 @@
-                insert->setColumn(table_idx, str);
-                break;
-            }
-            case AVRO_BYTES:
-            {
-                const void* s;
-                size_t len;
-                int r = avro_value_get_bytes(&field, &s, &len);
-                assert(r == 0);
-                insert->setColumn(table_idx, std::string((char*)s, len));
+            avro_value_t field;
+            avro_value_get_by_index(value, i, &field, NULL);
+            const char* name = avro_schema_record_field_name(avro_value_get_schema(value), i);
+            int table_idx = table.getColumn(name).getPosition();
 
-                break;
-            }
-            case AVRO_INT32:
-            {
-                int32_t out;
-                int r = avro_value_get_int(&field, &out);
-                assert(r == 0);
-                insert->setColumn(table_idx, out);
-                break;
-            }
-            case AVRO_INT64:
-            {
-                int64_t out;
-                int r = avro_value_get_long(&field, &out);
-                assert(r == 0);
-                insert->setColumn(table_idx, out);
-                break;
-            }
-            case AVRO_FLOAT:
-            {
-                float out;
-                int r = avro_value_get_float(&field, &out);
-                assert(r == 0);
-                insert->setColumn(table_idx, out);
-                break;
-            }
-            case AVRO_DOUBLE:
-            {
-                double out;
-                int r = avro_value_get_double(&field, &out);
-                assert(r == 0);
-                insert->setColumn(table_idx, out);
-                break;
-            }
-            case AVRO_BOOLEAN:
+            if ( avro_value_get_type(&field) == AVRO_UNION )
             {
-                int out = 0;
-                int r = avro_value_get_boolean(&field, &out);
-                assert(r == 0);
-                insert->setColumn(table_idx, (bool)out);
-                break;
+                int disc;
+                avro_value_get_discriminant(&field, &disc);
+         
+                if ( disc > 0 )
+                {
+                    avro_value_t union_value;
+                    avro_value_set_branch(&field, disc, &union_value);
+                    if( avro_value_get_type(&union_value) == AVRO_RECORD )
+                    {
+                        char* json_string;
+                        processAvroType_(insert, &union_value, table);
+                        // Put the raw JSON for a RECORD just in case
+                        avro_value_to_json(&union_value, 1, &json_string);
+                        std::string jsonString = std::string(json_string);
+                        insert->setColumn(table_idx, jsonString);
+                        free(json_string);
+                    }
+                    else
+                        processAvroNonRec(insert, &union_value, table, 
```

Comment:
Coding style in other files uses curly braces even for oneliners.
```suggestion
                    {
                        processAvroNonRec(insert, &union_value, table, table_idx);
                    }
```